homeassistant/components/group/entity.py

```diff
@@ -437,15 +450,38 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
             self._state = None
             return
         # If the entity domains have more than one
-        # on state, we use STATE_ON/STATE_OFF
+        # on state, we use STATE_ON/STATE_OFF, unless there is
+        # only one specific `on` state in use for one specific domain
+        elif (
+            len(active_on_states := _active_on_states()) == 1
+            and len(active_domains) == 1
+        ):
```

Comment:
I'd reverse this check since `len(active_domains)` is going to be far cheaper than enumeration the active on states